src/psyclone/psyir/symbols/symboltable.py

```diff
@@ -195,6 +195,42 @@ def __contains__(self, key):
         '''
         return key in self._symbols
 
+    def remove(self, key):
```

Comment:
Could the key be the symbol itself rather than the name, or do we need it to be the name>

Reply:
It could and now it is!